### PR TITLE
Make curl retry on transient errors

### PR DESCRIPTION
**Pull request by jhchabran**
Spotted some transient errors in our CI pipeline involving asdf plugins, solved by a trivial change so it's really just low-hanging fruit :)

### CODE REVIEW

**Review by filiphaftek — APPROVED** [review]
LGTM 👍

File: bin/list-all [hunk 1]
```diff
@@ -3,9 +3,9 @@
 releases_url="https://api.github.com/repos/mvdan/sh/releases"
 
 if [ -n "$GITHUB_API_TOKEN" ]; then
-	cmd="curl -s -H 'Authorization: token $GITHUB_API_TOKEN' $releases_url"
+	cmd="curl --retry 10 --retry-delay 2-s -H 'Authorization: token $GITHUB_API_TOKEN' $releases_url"
```
Comment (luizm) [inline bin/list-all]: missing a space here `2-s`

Comment (jhchabran) [reply bin/list-all]: @luizm apologies, I missed this notification. Just pushed the fix 🤦

Comment (luizm) [reply bin/list-all]: thank you ❤️ to be honest I forgot about this PR.. my bad

**Pull request merged**